src/bufr/DataObject.h

```diff
@@ -372,6 +399,131 @@ namespace Ingester
             typename std::enable_if<!std::is_arithmetic<T>::value, U>::type* = nullptr) const
         {
             throw std::runtime_error("The stored value was is not a number");
+            return 0.0f;
+        }
+
+        /// \brief Set the data associated with this data object (numeric DataObject).
```

Comment:
What's the purpose of this new block of code?

Reply:
@rmclaren should we move forward and strip out if not needed

or even better this is something needed and useful, right?